### DISCUSSION TITLE
Video Looping

### DISCUSSION THREAD

**Discussion by jh-martin** [gopro/labs#1014] [Q&A]
I want to adjust the 4K resolution frame rate to 60 and the longest loop duration to the preset loop mode of the black10 camera through QR code. How can I achieve this? thank you!

**Comment by dnewman-gpsw** [comment]
There is no 4K at 60 fps loop mode, so there is not QR Command to activate that mode. 4K30 or 1080p60 or 1080p120 are the options.

**Comment by jh-martin** [comment]
Thank you for your reply! May I ask if you can add the option of 4K60 and infinite loop duration in Labs? I need it. thank you!

**Comment by jh-martin** [comment]
Alternatively, could we add an option in Labs to reduce the split file size (such as 1G), which would be very useful when I use it with my dash cam.

**Reply by dnewman-gpsw** [reply]
Labs doesn't add new video modes. You can only modify existing ones, so it would not be possible to make a 4k60 looping mode.  However, you can set the chapter size to anything you like. That is an existing feature, the 64BT command. *64BT=1000 would be a 1 GB chapter size.

**Comment by jh-martin** [comment]
我可以将black10视频分段大小设置为任何我喜欢的大小吗（比如我录制了一个20G的视频，我设置了分段大小是1G，那么这个视频文件将会分为20个1G的文件，是这样吗？）？如果是这样，请把设置页面链接地址给我！非常感谢您的回复！
Can I set the black10 video segment size to any size I like (for example, if I record a 20GB video and set the segment size to 1G, then this video file will be divided into 20 1GB files, is that correct?)? If that's the case, please give me the link address for the settings page! Thank you very much for your reply!

**Reply by dnewman-gpsw** [reply]
It is https://gopro.github.io/labs/control/extensions/ see command ```64BT```

**Comment by jh-martin** [comment]
另外，给black10的视频添加日期和时间水印，任何分辨率和帧率都适用吗？谢谢您！
Also, is adding date and time watermarks to Black10 videos suitable for any resolution and frame rate? Thank you very much!

**Reply by dnewman-gpsw** [reply]
It is only above on 4K30 or lower.

**Comment by jh-martin** [comment]
Thank you very much for your reply! Unfortunately, time and date watermarks cannot be added to 4K/60 or higher. Looking forward to a new firmware update.

**Reply by dnewman-gpsw** [reply]
Correct, 4K 30 or lower.  No future firmware can address this.

**Comment by jh-martin** [comment]
Thank you very much! In theory, it is possible to add time and date to a video without distinguishing the video format, which is a reality in other motion cameras. I wonder what considerations GoPro is based on? However, based on 64BT and USB Power Trigger, it can meet my needs to implement the function of a dash cam, and I think I can solve the heat dissipation problem. Thank you again for your reply!

**Reply by dnewman-gpsw** [reply]
It is not a policy, it is a architecture issue for the way Labs renders text, no support beyond 4K 30.

**Comment by jh-martin** [comment]
Okay, got it. Thank you very much! After unplugging the battery, the QR code will become invalid and needs to be scanned again, right?

**Reply by dnewman-gpsw** [reply]
If you use the permanent option you don't need to rescan.

**Comment by jh-martin** [comment]
Hello! After scanning the USB Power Trigger with my black10, the QR code often becomes invalid. May I ask where to check the permanent mark? thank you!
QR URL: https://gopro.github.io/labs/control/usb/

**Reply by dnewman-gpsw** [reply]
That is already a permanent command, as it is a boot command, those have to be permanent by design, as that must run after a shutdown. How does you QR code become invalid?  The boot command is stored on the SD Card, so if you format or replace the SD card, that is the only way this will not continue to work.

**Comment by jh-martin** [comment]
Oh, that's right, I thought it was permanently stored in ROM. It failed after formatting the SD card. Thank you.

**Comment by jh-martin** [comment]
Hello! Why does my Black10 automatically stop recording for more than three hours without any prompts? I have tried several times and it's like pressing the recording button and the red light flashes a few times. We used the latest labs firmware and scanned three QR codes: USB Power Trigger, 64BT=1000, Video "Burn in" Overlays. thank you！
![20240816210423](https://github.com/user-attachments/assets/c5bdcf4e-f4ff-4650-92b1-5c59a1ec82ea)
![20240816210435](https://github.com/user-attachments/assets/d6ca3efc-065d-4838-8179-7ee9bd703da3)
![20240816211839](https://github.com/user-attachments/assets/17bc4940-90a1-44f4-bad8-c879d24fca2d)

**Comment by jh-martin** [comment]
Also, when I insert the battery, the camera automatically turns on. Can I set it to not automatically turn on when I insert the battery? Is this related to scanning QR Code: USB Power Trigger? thank you!

**Reply by dnewman-gpsw** [reply]
No, the battery is a power trigger, that can't be avoided.

I don't know why your camera stops recording. However, you are in experimental territory, as using a small 64TB, meaning you are using significantly more chapters per capture, and maybe you are hitting a chapter limit.

**Comment by jh-martin** [comment]
Thank you for your reply! I changed 64BT to 2048, but it still automatically stopped when there were 16 minutes left on the SD card, and there were no abnormal records in the qrlog.exe file. I don't know why! I'll try changing it to a larger number again. Thank you very much!

**Comment by jh-martin** [comment]
Hello! Yes, I changed it to a larger segmented chapter (! M64BT=0), which can be recorded continuously until the card is full. So, can we ask the engineer to modify the Labs so that it can also support continuous recording of smaller segmented chapters until the card is full (e.g. 60BT=1000)? Thank you!
